contrib/turbojpeg/tests/turbojpeg_sapi_test.cc

```diff
@@ -0,0 +1,374 @@
+};
+unsigned int image_len = 2412;
+// clang-format on
+
+#ifdef DEBUG_LOGGING
+#define MADE_IT                                                   \
+  do {                                                            \
+    std::cerr << "Reached " << __FILE__ << ':' << __LINE__ << ' ' \
+              << __PRETTY_FUNCTION__ << std::endl;                \
+  } while (0)
+#else
+#define MADE_IT \
+  do {          \
+  } while (0)
+#endif
+
+TEST_F(TurboJpegSapiSandboxTest, Compressor) {
+  static_assert(
+      std::is_same_v<decltype(api_->tjInitCompress()), absl::StatusOr<void*>>,
+      "bug");
+  absl::StatusOr<void*> compression_handle_raw = api_->tjInitCompress();
+  ASSERT_THAT(compression_handle_raw, IsOk());
+  ASSERT_THAT(compression_handle_raw.value(), NotNull());
+  sapi::v::RemotePtr compression_handle{compression_handle_raw.value()};
+
+  sapi::v::Array<unsigned char> array{sizeof image};
+  std::memcpy(array.GetData(), image, sizeof image);
+
+  sapi::v::GenericPtr buffer;
+  unsigned long actual_length;
+  {
+    sapi::v::ULong length{0};
+    auto result = api_->tjCompress2(
+        &compression_handle, array.PtrBefore(), 12, 36, 67, TJPF_RGB,
+        buffer.PtrAfter(), length.PtrBoth(), TJSAMP_444, 10,
+        0 * (TJFLAG_ACCURATEDCT | TJFLAG_PROGRESSIVE | TJFLAG_LIMITSCANS));
+    ASSERT_THAT(result, IsOk());
+    EXPECT_THAT(result.value(), Eq(0));
+    if (result.value()) {
+      auto errmsg = api_->tjGetErrorStr2(compression_handle.PtrNone());
+      ASSERT_THAT(errmsg, IsOk());
+      auto errstr =
+          sandbox_->GetCString(sapi::v::RemotePtr(errmsg.value()), 256);
+      ASSERT_THAT(errstr, IsOk());
+      ASSERT_THAT(errstr.value(), StrEq(""));
+      ASSERT_TRUE(false);
+    }
+    ASSERT_TRUE(buffer.GetValue());
+    ASSERT_TRUE(buffer.GetRemote());
+    ASSERT_TRUE((actual_length = length.GetValue()));
+  }
+  auto value = buffer.GetValue();
+  static_assert(std::is_same_v<decltype(value), uintptr_t>, "bug");
+
+  MADE_IT;
+
+  static_assert(
+      std::is_same_v<decltype(api_->tjInitDecompress()), absl::StatusOr<void*>>,
+      "bug");
+
+  MADE_IT;
+
+  auto result = api_->tjDestroy(&compression_handle);
+  ASSERT_THAT(result, IsOk());
+  ASSERT_THAT(result.value(), Eq(0));
+
+  MADE_IT;
+
+  absl::StatusOr<void*> decompression_handle_raw = api_->tjInitDecompress();
+  ASSERT_THAT(decompression_handle_raw, IsOk());
+  ASSERT_THAT(decompression_handle_raw.value(), NotNull());
+  sapi::v::RemotePtr decompression_handle{decompression_handle_raw.value()};
+
+  MADE_IT;
+
+  sapi::v::Int width{0}, height{0}, subsamp{0}, colorspace{0};
+  sapi::v::RemotePtr buffer_addr{reinterpret_cast<void*>(buffer.GetValue())};
+  result = api_->tjDecompressHeader3(
+      &decompression_handle, &buffer_addr, actual_length, width.PtrAfter(),
+      height.PtrAfter(), subsamp.PtrAfter(), colorspace.PtrAfter());
+  ASSERT_THAT(result, IsOk());
+  EXPECT_THAT(result.value(), Eq(0));
+
+  MADE_IT;
+
+  if (result.value()) {
```

Comment:
This will always be true (due to the `ASSERT_THAT`) and the test will always fail.